### ISSUE TITLE
Video wobbles if window is resized while playing

### ISSUE THREAD

**Issue by svobs** [iina/iina#4667]
**System and IINA version:**

- macOS 14.0 Sonoma
- IINA 1.3.3

**Expected behavior:**
Smooth animation during trackpad pinch-to-zoom

_Edit: also occurs when resizing window by any means while video is playing._

**Actual behavior:**

https://github.com/iina/iina/assets/2213815/7c2c73a9-e267-44c2-b751-234673783d93

https://github.com/iina/iina/assets/2213815/a4dcdb84-c66d-4dd5-9646-79008f79f55f

<br>

**Steps to reproduce:**
1. Go to IINA's `Settings` > `Control` > set `Pinch to:` to `Adjust window size`
2. Open a video in IINA and begin playing it.
3. Pinch the video. (Seems more prevalent on lower-FPS but high-quality videos. I picked a bad one to show above because it's 60fps) and I was playing it on a 60Hz display.

- [x] MPV does not have this problem.

**How often does this happen?**

Usually.

**Comment by svobs** [comment]
I spent a lot of time hunting for this one, but [this 11-year-old comment](https://stackoverflow.com/a/9934931/1347529) was the key, and happy that it ended up being a simple flag which was set incorrectly! Learned a lot more about `CAOpenGLLayer` in the process...

Did some testing and confirmed that the fix didn't alter the frame rate by more than 1 frame per sec during the animation, and made no impact to the framerate when not animating. I did find that it introduced a new race condition which could cause a crash during quit, but was able to fix it by checking for simple `Bool` flags without needing a lock.
